crates/h264/README.md

```diff
@@ -7,7 +7,231 @@
+
+// forbidden zero bit must be unset
+let _ = writer.write_bit(false);
+// nal_ref_idc is 0
+let _ = writer.write_bits(0, 2);
+// nal_unit_type must be 7
+let _ = writer.write_bits(7, 5);
+
+// profile_idc = 100
+let _ = writer.write_bits(100, 8);
+// constraint_setn_flags all false
+let _ = writer.write_bits(0, 8);
+// level_idc = 0
+let _ = writer.write_bits(0, 8);
+
+// seq_parameter_set_id is expg
+let _ = writer.write_exp_golomb(0);
+
+// branch to sps ext
+// chroma_format_idc is expg
+let _ = writer.write_exp_golomb(0);
+// bit_depth_luma_minus8 is expg
+let _ = writer.write_exp_golomb(0);
+// bit_depth_chroma_minus8 is expg
+let _ = writer.write_exp_golomb(0);
+// qpprime
+let _ = writer.write_bit(false);
+// seq_scaling_matrix_present_flag
+let _ = writer.write_bit(false);
+
+// return to sps
+// log2_max_frame_num_minus4 is expg
+let _ = writer.write_exp_golomb(0);
+// pic_order_cnt_type is expg
+let _ = writer.write_exp_golomb(0);
+// log2_max_pic_order_cnt_lsb_minus4 is expg
+let _ = writer.write_exp_golomb(0);
+
+// max_num_ref_frames is expg
+let _ = writer.write_exp_golomb(0);
+// gaps_in_frame_num_value_allowed_flag
+let _ = writer.write_bit(false);
+// 3840 width:
+// 3840 = (p + 1) * 16 - 2 * offset1 - 2 * offset2
+// we set offset1 and offset2 to both be 0 later
+// 3840 = (p + 1) * 16
+// p = 239
+let _ = writer.write_exp_golomb(239);
+// we want 2160 height:
+// 2160 = ((2 - m) * (p + 1) * 16) - 2 * offset1 - 2 * offset2
+// we set offset1 and offset2 to both be 0 later
+// m is frame_mbs_only_flag which we set to 1 later
+// 2160 = (2 - 1) * (p + 1) * 16
+// 2160 = (p + 1) * 16
+// p = 134
+let _ = writer.write_exp_golomb(134);
+
+// frame_mbs_only_flag
+let _ = writer.write_bit(true);
+
+// direct_8x8_inference_flag
+let _ = writer.write_bit(false);
+// frame_cropping_flag
+let _ = writer.write_bit(false);
+
+// vui_parameters_present_flag
+let _ = writer.write_bit(true);
+
+// enter vui to set the framerate
+// aspect_ratio_info_present_flag
+let _ = writer.write_bit(true);
+// we want square (1:1) for 16:9 for 4k w/o overscan
+// aspect_ratio_idc
+let _ = writer.write_bits(1, 8);
+
+// overscan_info_present_flag
+let _ = writer.write_bit(true);
+// we dont want overscan
+// overscan_appropriate_flag
+let _ = writer.write_bit(false);
+
+// video_signal_type_present_flag
+let _ = writer.write_bit(false);
+// chroma_loc_info_present_flag
+let _ = writer.write_bit(false);
+
+// timing_info_present_flag
+let _ = writer.write_bit(true);
+// we can set this to 100 for example
+// num_units_in_tick is a u32
+let _ = writer.write_bits(100, 32);
+// fps = time_scale / (2 * num_units_in_tick)
+// since we want 144 fps:
+// 144 = time_scale / (2 * 100)
+// 28800 = time_scale
+// time_scale is a u32
+let _ = writer.write_bits(28800, 32);
+let _ = writer.finish();
+
+// Now result contains a complete SPS.
+let result = Sps::parse(sps.into()).unwrap();
+```
+For more SPS examples, check out the tests in the source code for SPS.
```

Comment:
We dont want to use this lib to write the sps, this is only for parsing the sps.